src/components/Menu/Menu.js

```diff
@@ -0,0 +1,38 @@
+import { Button, Pressable, Text, View, TouchableOpacity, TouchableHighlight, ImageBackground, Image, Modal } from "react-native";
+import { useState } from "react";
+import SettingsModal from "./components/SettingsModal/SettingsModal";
+import styles from "./styles";
+
+const background = require("../../../src/img/backgrounds/menu-background-1.jpg");
+const settingIcon = require("../../img/icons/gear-3.png");
+
+const Menu = ({ navigation }) => {
+  const [modalVisible, setModalVisible] = useState(false);
```

Comment:
you may implement a custom hook for such cases (like `const [opened, open. close] = useOpenClose()`)